core/vm/lightclient/types.go

```diff
@@ -225,11 +227,11 @@ func (kvmp *KeyValueMerkleProof) Validate() bool {
 	kp = kp.AppendKey(kvmp.Key, merkle.KeyEncodingURL)
 
```

Comment:
How about adding all the check logic in the `Validate()` function?
Thus we don't need to change tendermint dep. The drawback is we have to decode the proof twice.

```suggestion

	decodedProof, err := prt.DecodeProof(kvmp.Proof)
	if err != nil {
		return false
	}
	for _, op := range decodedProof{
		if valueOp, ok := op.(iavl.IAVLValueOp); ok {
			if len(valueOp.Proof.Leaves) != 1 {
				return false
			}
			for _, innerNodde := range valueOp.Proof.LeftPath {
				if len(innerNodde.Right) > 0 && len(innerNodde.Left) > 0 {
					return false
				}
			}
		}
	}
```

Reply:
ok, I prefer to avoid `decode the proof twice`.